src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -0,0 +1,115 @@
+// (C) 2007-2019 GoodData Corporation
+import * as React from "react";
+import isEqual = require("lodash/isEqual");
+import noop = require("lodash/noop");
+import mapboxgl = require("mapbox-gl");
+import { DEFAULT_PUSHPIN_LOCATION } from "../../../constants/geoChart";
+import { IChartConfig } from "../../../interfaces/Config";
+
+import "../../../../styles/scss/geoChart.scss";
+
+const MAPBOX_ACCESS_TOKEN =
+    "pk.eyJ1IjoiaW1udXR6IiwiYSI6ImNrMHAxY2UxZzBnc2EzZG11YmVhd2dubG0ifQ.bUTN7ceAHq6kVooe3MKgqg";
+const MAPBOX_STYLE = "mapbox://styles/mapbox/streets-v11";
+mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+
+export interface IChartProps {
+    config: IChartConfig;
+    domProps: any;
+    callback(): void;
+}
+
+export interface IChartState {
+    center: [number, number];
+    zoom: number;
+}
+
+export default class GeoChartRenderer extends React.Component<IChartProps, IChartState> {
+    public static defaultProps: Partial<IChartProps> = {
+        callback: noop,
+        domProps: {},
+    };
+
+    private chart: mapboxgl.Map;
+    private chartRef: HTMLElement;
+
+    public constructor(props: IChartProps) {
+        super(props);
+
+        const {
+            config: { chart },
+        } = props;
+        const center =
+            chart && chart.center
+                ? chart.center
+                : [DEFAULT_PUSHPIN_LOCATION.LNG, DEFAULT_PUSHPIN_LOCATION.LAT];
```

Comment:
please refer to this [comment](https://github.com/gooddata/gooddata-react-components/pull/1307#discussion_r362358303)